test/reflection.jl

```diff
@@ -777,6 +777,7 @@ faz1(x) = foo1(x)
 @test faz1(1.0) == 1
 m = first(methods(bar1, Tuple{Int}))
 Base.delete_method(m)
```

Comment:
Does this mean that the result of `Base.delete_method` (and `Base.delete_binding`) won’t take effect until the user explicitly calls `Core.@worldinc`? That sounds like it would be a pretty breaking change. I agree with the idea of restricting the effect of `:worldinc` to the top level (for helping inference), but it seems like `Base.delete_method` might need some adjustments, such as calling `@eval` internally or something similar.

Reply:
> Does this mean that the result of Base.delete_method (and Base.delete_binding) won’t take effect until the user explicitly calls Core.@worldinc? 

It takes effect immediately (in a new world age), but the effect is not visible in the current task until it raises it world age.

Calling `@eval` internally will not raise the world age, because it's not at top level. `delete_method` is mostly used internally by Revise which can take an appropriate adjustment. There could also be a `delete_method` macro that does the world age raising implicitly.